lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -25,7 +28,20 @@ function replaceAsString(
   }
   // Look for the old version number
   const searchString = `"${oldValue}"`;
-  const newString = `"${newValue}"`;
+  let newString = `"${newValue}"`;
+
+  if (oldValue.match(patchReg)) {
+    //todo tests
+    // "patch:lodash@npm:4.16.0#patches/lodash.patch".replace(new RegExp("(patch:lodash@(npm:)?[><=~^]+).*#"), "$1"+newValue+"#");
+    // "patch:metro@0.58.0#./.patches/metro.patch".replace(new RegExp("(patch:metro@(npm:)?[><=~^]+).*#"), "$1"+newValue+"#";
+    // "patch:metro@^0.58.0#./.patches/metro.patch".replace(new RegExp("(patch:metro@(npm:)?[><=~^]+).*#"), "$1"+newValue+"#");
+    const regex = new RegExp('(patch:' + depName + '@(npm:)?).*#');
```

Comment:
```suggestion
    const regex = new regEx('(patch:' + depName + '@(npm:)?).*#');
```
Use our regex function

Reply:
i'm trying but i'm not familiar with this regEx and for some reason whatever i do it doesnt work
`   const regex = regEx('(patch:.*@(npm:)?).*#');` ////// THIS WORKS
but when i try to do it with parameters, by copying the way from other regEx.. it doesnt work, 
what am i missing?
`    const regex = regEx(/^(patch:(?<depName>@(npm:)?)).*#'$/);`

Reply:
You probably need to escape special chars? Re2 is more strict in those areas